lib/cn/csched_sp3.c

```diff
@@ -1104,7 +1105,22 @@ sp3_process_workitem(struct sp3 *sp, struct cn_compaction_work *w)
         }
     }
 
-    sp3_dirty_node_locked(sp, tn);
+    if (split) {
```

Comment:
Why not follow the existing pattern?  if (w->cw_action == CN_ACTION_SPLIT) {

Reply:
Yes, will do. I had this check in more than one place in the test version of this code and added a bool for it.. Forgot to remove the bool when reverting the test code.